### PR TITLE
[FLINK-22934][docs] Add instructions for using the " ' " escape syntax of SQL client

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

*Add instructions for using the " ' " escape syntax of SQL client*

## Brief change log

*Add instructions for using the " ' " escape syntax of SQL client*

## Verifying this change

*Add instructions for using the " ' " escape syntax of SQL client*

*A pure documentation work.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/content.zh/docs/dev/table/sqlClient.md [hunk 1]
```diff
@@ -153,6 +153,37 @@ Received a total of 5 rows
...
+| name2  |   2 |
++--------+-----+
+2 rows in set
+```
+
+为了查询出 name 列值为 `"n'ame1"` 的记录，你应该使用：
+
+```sql
+select * from t where name='n''ame1'
+-- or
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/sqlClient.md:175]:
```suggestion
-- 或者
```

File: docs/content.zh/docs/dev/table/sqlClient.md [hunk 2]
```diff
@@ -153,6 +153,37 @@ Received a total of 5 rows
...
+select * from t where name='n''ame1'
+-- 或者
+select * from t where name like 'n''ame1'
+```
+
+而不是
+
+```text
+select * from t where name='n\'ame1'
+-- or
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/sqlClient.md:183]:
```suggestion
-- 或者
```

**Comment by wuchong** [comment]
Thanks for the contribution @RocMarshal .  This escape syntax is not only supported in SQL Client, but all the table-ecosystem including TableEnvironment. So it would be better to put the content in https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/table/sql/queries/overview/#syntax

File: docs/content.zh/docs/dev/table/sql/queries/overview.md [hunk 3]
```diff
@@ -393,7 +393,37 @@ Flink SQL uses a lexical policy for identifier (table, attribute, function names
 - After which, identifiers are matched case-sensitively.
 - Unlike Java, back-ticks allow identifiers to contain non-alphanumeric characters (e.g. <code>"SELECT a AS `my field` FROM t"</code>).
 
-String literals must be enclosed in single quotes (e.g., `SELECT 'Hello World'`). Duplicate a single quote for escaping (e.g., `SELECT 'It''s me.'`). Unicode characters are supported in string literals. If explicit unicode code points are required, use the following syntax:
+String literals must be enclosed in single quotes (e.g., `SELECT 'Hello World'`). Duplicate a single quote for escaping (e.g., `SELECT 'It''s me.'`).
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/overview.md:396]: I think a simple example is enough: 

```
Flink SQL> SELECT 'Hello World', 'It''s me';
+-------------+---------+
|      EXPR$0 |  EXPR$1 |
+-------------+---------+
| Hello World | It's me |
+-------------+---------+
1 row in set
```

Comment (RocMarshal) [reply docs/content.zh/docs/dev/table/sql/queries/overview.md]: @wuchong  Thank you for your suggestions. I make some changes based on your comments. Please take a look.

**Review by wuchong — APPROVED** [review]
Thanks for the updating. LGTM.

**Comment by RocMarshal** [comment]
re-run the last Azure build

**Comment by wuchong** [comment]
@flinkbot run azure

**Pull request merged**